lib/chef/knife/core/windows_bootstrap_context.rb

```diff
@@ -276,7 +260,7 @@ def install_chef
         end
 
         def bootstrap_directory
-          cleanpath(ChefConfig::Config.etc_chef_dir(true))
+          "C:\\chef"
```

Comment:
that probably shouldn't be hardcoded

Reply:
Okay, so this needs to simply be a call to `ChefConfig::Config.etc_chef_dir(true)` and the cleanpath() around it needs to be dropped.

What needs to be fixed here is the actual implementation of `etc_chef_dir` and all the rest of the helpers:

https://github.com/chef/chef/blob/master/chef-config/lib/chef-config/config.rb#L77-L81

The `PathHelper.cleanpath()` method needs to take the same boolean `is_windows` argument with the same default which then controller if `cleanpath` behaves like windows or not.

And the reference to `File::SEPARATOR` needs to be removed.

That needs to be a call to `file_separator`:

https://github.com/MsysTechnologiesllc/chef/blob/dh/fix-windows-to-windows-bootstrap/chef-config/lib/chef-config/path_helper.rb#L50-L56

But that method signature needs to change to take a `(is_windows = ChefUtils.windows?)` and then the `File::SEPARATOR` references need to be changed to be the right constants based on is_windows (they should likely no longer reference `File::` at all).

Reply:
In progress to implement these comments.

Reply:
@lamont-granquist Updated the recommended changes but there are definitely some suggestions/changes needed I assume. I am making the changes in suggestion go by manner here.